packages/sdk/src/machines/post-ram-machine/index.ts

```diff
@@ -0,0 +1,126 @@
+import { assign, fromPromise, setup } from "xstate";
+import { ArtBlocksClient } from "../..";
+import {
+  BidDetailsFragment,
+  GetUserBidsDocument,
+} from "../../generated/graphql";
+import { ProjectDetails } from "../project-sale-manager-machine/utils";
+import { getMessageFromError } from "../utils";
+
+export type PostRAMMachineEvents = {
+  type: "RESET";
+};
+
+export type PostRAMMachineContext = {
+  // Client used to interact with ArtBlocks API with access to walletClient and publicClient
+  artblocksClient: ArtBlocksClient;
+  // Details of the project relevant to sale
+  project: NonNullable<ProjectDetails>;
+  // Error message that may arise during the bidding process
+  errorMessage?: string;
+  // List of user bids for the current project
+  userBids: Array<BidDetailsFragment>;
+};
+
+export type PostRAMMachineInput = Pick<
+  PostRAMMachineContext,
+  "artblocksClient" | "project"
+>;
+
+export const postRAMMachine = setup({
+  types: {
+    input: {} as Pick<PostRAMMachineContext, "project" | "artblocksClient">,
+    context: {} as PostRAMMachineContext,
+    events: {} as PostRAMMachineEvents,
+  },
+  actors: {
+    fetchUserBids: fromPromise(
+      async ({
+        input,
+      }: {
+        input: Pick<PostRAMMachineContext, "artblocksClient" | "project">;
```

Comment:
same `PostRAMMachineInput`?

Reply:
This one I think  I'd actually prefer to leave as is because PostRAMMachineInput is specifically for the input required to spawn the machine and this is input to a specific actor.